src/js/containers/agencyV2/visualizations/ObligationsByAwardTypeContainer.jsx

```diff
@@ -43,79 +44,85 @@ export default function ObligationsByAwardTypeContainer({ fiscalYear, windowWidt
-                {
-                    label: ['Financial', 'Assistance'], // line break between words
-                    value: 0,
-                    color: '#C05600'
-                },
-                {
-                    label: ['Contracts', ''], // so each cat label array is same length
-                    value: 0,
-                    color: '#545BA3'
-                }
-            ];
-            const details = [
-                {
-                    label: ['Grants'],
-                    color: '#E66F0E'
-                },
-                {
-                    label: ['Loans'],
-                    color: '#FFBC78'
-                },
-                {
-                    label: ['Direct Payments'],
-                    color: '#FA9441'
-                },
-                {
-                    label: ['Other'],
-                    color: '#FCE2C5'
-                },
-                {
-                    label: ['Contracts'],
-                    color: '#7F84BA'
-                },
-                {
-                    label: ['IDVs'],
-                    color: '#A9ADD1'
-                }
-            ];
-            res.data.results.forEach((d) => {
-                switch (d.category) {
-                    case 'grants':
-                        categories[0].value += d.aggregated_amount;
-                        details[0].value = d.aggregated_amount;
-                        break;
-                    case 'loans':
-                        categories[0].value += d.aggregated_amount;
-                        details[1].value = d.aggregated_amount;
-                        break;
-                    case 'direct_payments':
-                        categories[0].value += d.aggregated_amount;
-                        details[2].value = d.aggregated_amount;
-                        break;
-                    case 'other':
-                        categories[0].value += d.aggregated_amount;
-                        details[3].value = d.aggregated_amount;
-                        break;
-                    case 'contracts':
-                        categories[1].value += d.aggregated_amount;
-                        details[4].value = d.aggregated_amount;
-                        break;
-                    case 'idvs':
-                        categories[1].value += d.aggregated_amount;
-                        details[5].value = d.aggregated_amount;
-                        break;
-                    default:
-                        console.error(`Category name from API not recognized: ${d.category}`);
-                        setError(true);
-                }
-            });
-            setCategoriesForGraph(categories);
-            setDetailsForGraph(details);
-            setLoading(false);
-            obligationsByAwardTypeRequest.current = null;
+            if (Object.keys(res.data).length === 0) {
+                setNoData(true);
+                setLoading(false);
```

Comment:
Can we add `obligationsByAwardTypeRequest.current = null;` below setLoading

Reply:
done